packages/dashboard/src/components/Form/SelfServiceForm.js

```diff
@@ -1,87 +1,68 @@
+import * as React from "react";
 import { useFormik, getIn, setIn } from "formik";
 import classnames from "classnames";
 import SelfServiceMessages from "./SelfServiceMessages";
 
-export default function SelfServiceForm({ flow, config, fieldsConfig, title, button = "Submit" }) {
-  const fields = config.fields
-    .filter((field) => !field.name.startsWith("traits.name")) // drop name fields
-    .map((field) => ({ ...field, ...fieldsConfig[field.name] }))
-    .sort((a, b) => (a.position < b.position ? -1 : 1));
+export default function SelfServiceForm({ fieldsConfig, onSubmit, title, validationSchema = null, button = "Submit" }) {
+  const [messages, setMessages] = React.useState([]);
+  const fields = fieldsConfig.sort((a, b) => (a.position < b.position ? -1 : 1));
   const formik = useFormik({
     initialValues: fields.reduce((acc, field) => setIn(acc, field.name, field.value ?? ""), {}),
+    validationSchema,
+    onSubmit: async (values) => {
+      if (!formik.isValid) return;
```

Comment:
`nit` I dislike these inline returns that do not adhere to the overall code style